File: php/src/Advertisement/Application/Errors/AdvertisementNotFoundError.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\Application\Errors;

use Demo\App\Common\Result;

final readonly class AdvertisementNotFoundError
{
    public static function build(string $id): Result
    {
        return Result::failure('Advertisement not found with ID: ' . $id, 'NOT_FOUND');
    }
}

File: kotlin/src/main/kotlin/advertisement/domain/errors/AdvertisementNotFoundError.kt
package advertisement.domain.errors

import common.domain.EntityNotFoundDomainError

open class AdvertisementNotFoundError private constructor(errorMessage: String): EntityNotFoundDomainError(errorMessage) {
    companion object {
        fun withId(value: String): AdvertisementNotFoundError {
            val errorMessage = "Advertisement not found with Id $value"

            return AdvertisementNotFoundError(errorMessage)
        }
    }
}
